docs/plugins/data_persistence.md

```diff
@@ -0,0 +1,232 @@
+# Data persistence
+There are four plugins that currently have data persistence ability in opensearch dashboard: dashboard, discover, timeline, visualize, and vis-builder. They are using the following services and mechanisms from opensearch_dashboard_utils plugin to achieve data persistence.
+
+State syncing utils are a set of helpers to sync application state with URL or browser storage(when turning state: `storeInSessionStore` on in advanced setting, in case of overflowed URL):
+1. `syncState()`: subscribe to state changes and push them to state storage; subscribe to state storage and push them to state container
+2. storages that are compatible with `syncState()`
+* `OsdUrlStateStorage`: serialize state and persist it to URL's query param in rison format; listen for state change in URL and update them back to state
+* `SessionStorageStateStorage`: serialize state and persist it to URL's query param in session storage
```

Comment:
Is this the intended list formatting, or should these also be numbered and indented?

Reply:
Either works. Changed to numbered and indented for better readability.